server.go

```diff
@@ -84,40 +92,37 @@ func handleRequest(conn net.Conn) {
 		sendSessionMsg(enc, ethrMsg)
 		return
 	}
-	cleanupFunc := func() {
+	defer func() {
 		test.ctrlConn.Close()
 		close(test.done)
 		deleteTest(test)
-	}
+	}()
 	ui.emitTestHdr()
 	if test.testParam.TestId.Type == Pps {
 		err = runServerPpsTest(test)
 		if err != nil {
-			cleanupFunc()
+			ui.printMsg(fmt.Sprintf("run server pps test: %v",err))
 			return
 		}
 	}
 	ethrMsg = createAckMsg()
 	err = sendSessionMsg(enc, ethrMsg)
 	if err != nil {
-		cleanupFunc()
+		ui.printMsg(fmt.Sprintf("send session message: %v",err))
 		return
 	}
 	ethrMsg = recvSessionMsg(dec)
 	if ethrMsg.Type != EthrAck {
-		cleanupFunc()
 		return
 	}
 	test.isActive = true
 	var b [1]byte
 	_, err = test.ctrlConn.Read(b[0:])
 	ui.printMsg("Ending " + testToString(testParam.TestId.Type) + " test from " + server)
 	test.isActive = false
-	cleanupFunc()
```

Comment:
The reason for not using defer for cleanup in this code was so that ui.emitTestHdr() all is executed after control connection is closed and running goroutines exit. Without that, we may see messages from previous running test after emitTestHdr and it makes output harder to read.